component/src/main/java/org/wso2/extension/siddhi/map/avro/sourcemapper/AvroSourceMapper.java

```diff
@@ -0,0 +1,424 @@
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case DOUBLE:
+                                if (JsonToken.VALUE_NUMBER_FLOAT.equals(jsonToken)) {
+                                    data[position] = parser.getValueAsDouble();
+                                } else {
+                                    log.error("Avro message " + avroMessage +
+                                            " contains incompatible attribute types and values. Value " +
+                                            parser.getText() + " is not compatible with type DOUBLE. " +
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case STRING:
+                                data[position] = parser.getValueAsString();
+                                break;
+                            case FLOAT:
+                                if (JsonToken.VALUE_NUMBER_FLOAT.equals(jsonToken) ||
+                                        JsonToken.VALUE_NUMBER_INT.equals(jsonToken)) {
+                                    data[position] = attributeConverter.getPropertyValue(parser.getValueAsString(),
+                                            Attribute.Type.FLOAT);
+                                } else {
+                                    log.error("Avro message " + avroMessage +
+                                            " contains incompatible attribute types and values. Value " +
+                                            parser.getText() + " is not compatible with type FLOAT. " +
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case LONG:
+                                if (JsonToken.VALUE_NUMBER_INT.equals(jsonToken)) {
+                                    data[position] = parser.getValueAsLong();
+                                } else {
+                                    log.error("Json message " + avroMessage +
+                                            " contains incompatible attribute types and values. Value " +
+                                            parser.getText() + " is not compatible with type LONG. " +
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case OBJECT:
+                                switch (jsonToken) {
+                                    case START_OBJECT:
```

Comment:
It will check on START_ARRAY case and the START_ARRAY case has a break. This is done because the START_ARRAY and START_OBJECT cases need to have the same implementation.